serverless/networking/serverless-ossm-tls.adoc

```diff
@@ -0,0 +1,165 @@
+. Create TLS certificates:
++
+
+[source,terminal]
+----
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=Example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+----
+
++
+
+[source,terminal]
+----
+$ openssl req -out custom.example.com.csr -newkey rsa:2048 -nodes -keyout custom.example.com.key -subj "/CN=custom-ksvc-domain.example.com/O=Example Inc."
+----
+
++
+
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in custom.example.com.csr -out custom.example.com.crt
+----
+
+. Check that the certificates appear in your directory:
++
+
+[source,terminal]
+----
+$ ls -1
+----
+
++
+
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+
+. Create a secret:
++
+
+[source,terminal]
+----
+$  oc create -n istio-system secret tls custom.example.com --key=custom.example.com.key --cert=custom.example.com.crt
+----
+
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+
+[source,yaml]
+----
+istio-ingressgateway:
+  secretVolumes:
+  - mountPath: /custom.example.com
+    name: custom-example-com
+    secretName: custom.example.com
+----
+
+. Update the Istio ingress gateway to use your secret.
+.. Edit the `default-gateway` resource:
++
+
+[source,terminal]
+----
+$ oc edit gateway default-gateway
+----
+
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+
+[source,yaml]
+----
+- hosts:
+  - custom-ksvc-domain.example.com
+  port:
+    name: https
+    number: 443
+    protocol: HTTPS
+  tls:
+    mode: SIMPLE
+    privateKey: /custom.example.com/tls.key
+    serverCertificate: /custom.example.com/tls.crt
+----
+
+. Update the route to use pass-through TLS and `8443` as the `spec.port.targetPort`.
+.. Edit the route:
++
+
+[source,terminal]
+----
+$ oc edit route -n istio-system hello
+----
+
+. Add the following configuration to the route:
++
+
+[source,yaml]
+----
+spec:
+  host: custom-ksvc-domain.example.com
+  port:
+    targetPort: 8443
+  tls:
+    insecureEdgeTerminationPolicy: None
+    termination: passthrough
+  to:
+    kind: Service
+    name: istio-ingressgateway
+    weight: 100
+  wildcardPolicy: None
+----
+
+.Verification steps
+
+. Access your serverless application by secure connection:
++
+
+[source,terminal]
+----
+$ curl --cacert example.com.crt -H "Host: custom-ksvc-domain.example.com" --resolve "custom-ksvc-domain.example.com:443:<ip_address>" https://custom-ksvc-domain.example.com:443
+----
+
++
+where `<ip_address>` is the IP address that the {product-title} ingress router is exposed to.
```

Comment:
This is totally not clear how to get such IP address. It should be either documented or linked to a place where someone could understand how to get this address.

Reply:
We can add the following command: 

---------
For example, you can find the IP address by the following command.

```
oc get svc -n openshift-ingress router-default -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
```
---------

Reply:
Just one note, cluster on AWS exposes the `.status.loadBalancer.ingress[0].hostname` instead of `.status.loadBalancer.ingress[0].ip`. So AWS env does not work the command `oc get svc -n openshift-ingress router-default -o jsonpath='{.status.loadBalancer.ingress[0].ip}'`.

But I think we just can say "For example" and it should be fine without covering all env.

Reply:
I don't think we should add this if it's not for all environments, since OCP docs in general apply for all of these:
![image](https://user-images.githubusercontent.com/40172997/94715022-89bb9a00-0312-11eb-8d1c-db46d5e3fb4f.png)

To me it's just opening a can of worms where folks will try to use this command and then complain that it doesn't work for AWS. Is there nowhere in service mesh or istio docs for example that this is already covered in more depth that we can point to?

Reply:
ServiceMesh has a similar doc https://docs.openshift.com/container-platform/4.5/service_mesh/service_mesh_user_guide/ossm-traffic-manage.html#ossm-routing-determine-ingress_routing-traffic.
But actually it does not work at all :cry:  I opened the issue https://github.com/openshift/openshift-docs/issues/26032.

Meanwhile, it is better to keep using `where <ip_address> is the IP address that the {product-title} ingress router is exposed to.` ?

Reply:
I don't think so. Without your [comment](https://github.com/openshift/openshift-docs/pull/25706/files#r494763356) I would not be able to get this address. I remember I was getting it before, so switching to hostname wasn't a big deal for me.

Without such a hint, our users will have hard time to follow the docs. At least some of them, as I had.